components/org.wso2.carbon.identity.media.endpoint/src/main/java/org/wso2/carbon/identity/media/endpoint/service/MediaService.java

```diff
@@ -0,0 +1,242 @@
+            } else {
+                fileSecurity = new FileSecurity(false);
+            }
+        }
+        mediaMetadata.setFileSecurity(fileSecurity);
+    }
+
+    /**
+     * Validate if the uploaded media content type is a supported media type.
+     *
+     * @param mediaTypePathParam The media type available as a path parameter in the upload request.
+     * @param mediaType          The content type of the uploaded media.
+     */
+    public void validateMediaType(String mediaTypePathParam, MediaType mediaType) {
+
+        if (!mediaTypePathParam.equals(mediaType.getType())) {
+            throw handleException(Response.Status.FORBIDDEN,
+                    MediaServiceConstants.ErrorMessage.ERROR_CODE_ERROR_UPLOADING_MEDIA_CONTENT_TYPE_MISMATCH,
+                    mediaTypePathParam);
+        }
+
+        List<String> allowedContentTypes = loadAllowedContentTypes();
+        if (CollectionUtils.isEmpty(allowedContentTypes) || !allowedContentTypes.contains(mediaTypePathParam)) {
+            throw handleException(Response.Status.FORBIDDEN,
+                    MediaServiceConstants.ErrorMessage.ERROR_CODE_ERROR_UPLOADING_MEDIA_UNSUPPORTED_CONTENT_TYPE);
+        }
+
+        List<String> allowedContentTypeSubTypes = loadAllowedContentTypeSubTypes(mediaType.getType());
+        if (CollectionUtils.isEmpty(allowedContentTypeSubTypes) ||
+                !allowedContentTypeSubTypes.contains(mediaType.getSubtype())) {
+            throw handleException(Response.Status.FORBIDDEN,
+                    MediaServiceConstants.ErrorMessage.ERROR_CODE_ERROR_UPLOADING_MEDIA_UNSUPPORTED_CONTENT_TYPE);
+        }
+    }
+
+    /**
+     * Read allowed content types defined in media.properties file.
+     *
+     * @return list of allowed content types.
+     */
+    private List<String> loadAllowedContentTypes() {
+
+        Properties properties = new Properties();
+        try {
+            ClassLoader classLoader = this.getClass().getClassLoader();
+            if (classLoader != null) {
+                properties.load(Objects.requireNonNull(classLoader.getResourceAsStream(MEDIA_PROPERTIES_FILE_NAME)));
+                return Arrays.asList(properties.getProperty(ALLOWED_CONTENT_TYPES).split(","));
+            }
+        } catch (IOException e) {
+            MediaServiceConstants.ErrorMessage errorMessage =
+                    MediaServiceConstants.ErrorMessage.ERROR_CODE_ERROR_LOADING_ALLOWED_CONTENT_TYPES;
+            Response.Status status = Response.Status.INTERNAL_SERVER_ERROR;
+            throw handleException(e, errorMessage, LOG, status, MEDIA_PROPERTIES_FILE_NAME);
+        }
+        return null;
+    }
+
+    /**
+     * Read allowed sub types for a given content type in media.properties file.
+     *
+     * @return list of allowed sub content types for a given content type.
+     */
+    private List<String> loadAllowedContentTypeSubTypes(String contentType) {
```

Comment:
Can we change this name to `loadAllowedSubContentTypes` ?

Reply:
changed it as `loadAllowedContentSubTypes`